docs/specification/draft/basic/utilities/tasks.mdx

```diff
@@ -0,0 +1,808 @@
+---
+title: Tasks
+---
+
+<div id="enable-section-numbers" />
+
+<Info>**Protocol Revision**: draft</Info>
+
+<Note>
+
+Tasks are newly-introduced in this version of the MCP specification and are currently considered **experimental**.
+The design and behavior of tasks may evolve in future protocol versions.
+
+</Note>
+
+The Model Context Protocol (MCP) allows requestors (which can be either clients or servers, depending on the direction of communication) to augment their requests with **tasks**. Tasks are durable state machines that carry information about the underlying execution state of the request they wrap, and are intended for requestor polling and deferred result retrieval. Each task is uniquely identifiable by a receiver-generated **task ID**.
+
+## User Interaction Model
+
+Tasks are designed to be **application-driven** - receivers tightly control which requests (if any) support task-based execution and manage the lifecycles of those tasks; meanwhile, requestors are responsible for augmenting requests with tasks and for polling for the results of those tasks.
```

Comment:
Separate point - I think there's been a bunch of discussion about "who" decides whether a task is run sync or async. Ultimately we settled on the `requestor` (usually client) making this decision, and I think it's not immediately obvious why this choice was made (e.g. server has more information about how long a task might take, hence we added the `taskHint` field).

Should we add a sentence or two why this choice was made? Maybe something like this?

> This requestor-driven approach ensures deterministic response handling and enables sophisticated patterns like parallel task dispatch and coordinated result gathering, which only the requestor has sufficient context to orchestrate.